### PR TITLE
Freoduo - handwired split ortho keyboard.

### PR DESCRIPTION
**Pull request by FilipParyz**
I'd like to add my own handwired split ortho keyboard to the mix of handwired keyboards on QMK.
I've been testing it for quite some time now and everything looks to work well.

## Description
Handwired Split Ortho Keyboard with RGB underlight.
Two halves are connected with TRRS cable and each half contains one ProMicro Atmega32u4 microcontroller.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* NONE

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/freoduo/config.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x0616 //F is 6th in alphabet, P is 16th :)
+#define PRODUCT_ID      0x0602 //F is 6th in alphabet, 02 as in DUO
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    ParyzFilip
+#define PRODUCT         Freoduo
+#define DESCRIPTION     Handwired split ortholinear keyboard
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/config.h:26]:
```suggestion
```

File: keyboards/handwired/freoduo/freoduo.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    L30, L31, L32, L33, L34, L35, R30, R31, R32, R33, R34, R35, \
+    L40, L41, L42, L43, L44,           R41, R42, R43, R44, R45  \
+    ) \
+    ( \
+        KC_##L00, KC_##L01, KC_##L02, KC_##L03, KC_##L04, KC_##L05, KC_##R00, KC_##R01, KC_##R02, KC_##R03, KC_##R04, KC_##R05, \
+        KC_##L10, KC_##L11, KC_##L12, KC_##L13, KC_##L14, KC_##L15, KC_##R10, KC_##R11, KC_##R12, KC_##R13, KC_##R14, KC_##R15, \
+        KC_##L20, KC_##L21, KC_##L22, KC_##L23, KC_##L24, KC_##L25, KC_##R20, KC_##R21, KC_##R22, KC_##R23, KC_##R24, KC_##R25, \
+        KC_##L30, KC_##L31, KC_##L32, KC_##L33, KC_##L34, KC_##L35, KC_##R30, KC_##R31, KC_##R32, KC_##R33, KC_##R34, KC_##R35, \
+        KC_##L40, KC_##L41, KC_##L42, KC_##L43, KC_##L44,                     KC_##R41, KC_##R42, KC_##R43, KC_##R44, KC_##R45  \
+    )
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/freoduo.h:39]:
```suggestion
```

File: keyboards/handwired/freoduo/readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+A split 60%, 5x12 ortholinear keyboard handwired by Filip Pary�.
+
+Keyboard Maintainer: [Filip Paryz](https://github.com/FilipParyz)  
+Hardware Supported: Pro Micro, ATmega32u4
+
+Make example for this keyboard (after setting up your build environment):
+
+    make handwired/freoduo:default
+
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/readme.md:13]:
```suggestion
# Freoduo

A split 60%, 5x12 ortholinear keyboard handwired by Filip Paryż.

* Keyboard Maintainer: [Filip Paryz](https://github.com/FilipParyz)
* Hardware Supported: Pro Micro, ATmega32u4

Make example for this keyboard (after setting up your build environment):

    make handwired/freoduo:default

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/handwired/freoduo/rules.mk [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+VELOCIKEY_ENABLE = yes
+SPLIT_KEYBOARD = yes
+LAYOUTS = LAYOUT
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/rules.mk:34]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
VELOCIKEY_ENABLE = yes
SPLIT_KEYBOARD = yes
```

File: keyboards/handwired/freoduo/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+ * | Ctrl |ADJUST| Alt  | GUI  |    Space    |  Backspace  | Alt  | Left |Down  |Right |
+ * |      |      |      |      |             |             |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT( \
+  KC_GESC,            KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL, \
+  KC_TAB,             KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_ENT,  \
+  LT(_LOWER,KC_CAPS), KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, LT(_RAISE, KC_QUOT), \
+  KC_LSFT,            KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   RSFT_T(KC_SLSH),  \
+  KC_LCTL,            ADJUST,  KC_LGUI, KC_LALT, KC_SPC,                    KC_BSPC, KC_RALT, KC_LEFT, KC_DOWN, KC_RGHT  \
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/keymaps/default/keymap.c:87]:
```suggestion
[_QWERTY] = LAYOUT(
  KC_GESC,            KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
  KC_TAB,             KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_ENT,
  LT(_LOWER,KC_CAPS), KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, LT(_RAISE, KC_QUOT),
  KC_LSFT,            KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   RSFT_T(KC_SLSH),
  KC_LCTL,            ADJUST,  KC_LGUI, KC_LALT, KC_SPC,                    KC_BSPC, KC_RALT, KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/handwired/freoduo/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |      |      |      |             |             |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12, \
+  _______, _______, KC_UP,   _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+  _______, KC_LEFT, KC_DOWN, KC_RGHT, _______, _______, _______, KC_NUHS, KC_NUBS, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______  \
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/keymaps/default/keymap.c:108]:
```suggestion
[_LOWER] = LAYOUT(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  _______, _______, KC_UP,   _______, _______, _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
  _______, KC_LEFT, KC_DOWN, KC_RGHT, _______, _______, _______, KC_NUHS, KC_NUBS, _______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______
```

File: keyboards/handwired/freoduo/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |      |      |      |             |             |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT(
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,     KC_F9,     KC_F10,  KC_F11,  KC_F12, \
+  KC_DEL,  _______, _______, _______, _______, _______, _______, KC_UNDS,   KC_PLUS,   KC_LCBR, KC_RCBR, KC_PIPE,\
+  _______, _______, _______, _______, _______, _______, _______, S(KC_NUHS),S(KC_NUBS),_______, _______, _______,\
+  _______, _______, _______, _______, _______, _______, _______, _______,   _______,   _______, _______, _______,\
+  _______, _______, _______, _______, _______,                   _______,   _______,   _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/keymaps/default/keymap.c:129]:
```suggestion
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,     KC_F9,     KC_F10,  KC_F11,  KC_F12,
  KC_DEL,  _______, _______, _______, _______, _______, _______, KC_UNDS,   KC_PLUS,   KC_LCBR, KC_RCBR, KC_PIPE,
  _______, _______, _______, _______, _______, _______, _______, S(KC_NUHS),S(KC_NUBS),_______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______,   _______,   _______, _______, _______,
  _______, _______, _______, _______, _______,                   _______,   _______,   _______, _______, _______
```

File: keyboards/handwired/freoduo/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |Val Dn|Val Up|      |             |             |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] =  LAYOUT( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12, \
+  RESET  , RGB_TOG, RGB_MOD, VLK_TOG, _______, _______, _______, _______, _______, _______, _______, _______,\
+  _______, RGB_HUD, RGB_HUI, _______, _______, _______, _______, _______, _______, _______, _______, _______,\
+  _______, RGB_SAD, RGB_SAI, _______, _______, _______, _______, _______, _______, _______, _______, _______,\
+  _______, RGB_VAD, RGB_VAI, _______, _______,                   _______, _______, _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/keymaps/default/keymap.c:150]:
```suggestion
[_ADJUST] =  LAYOUT(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  RESET  , RGB_TOG, RGB_MOD, VLK_TOG, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, RGB_HUD, RGB_HUI, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, RGB_SAD, RGB_SAI, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, RGB_VAD, RGB_VAI, _______, _______,                   _______, _______, _______, _______, _______
```

File: keyboards/handwired/freoduo/info.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  "layouts": {
+    "LAYOUT": {
+      "layout": [{"x":0, "y":0}, {"x":1, "y":0}, {"x":2, "y":0}, {"x":3, "y":0}, {"x":4, "y":0}, {"x":5, "y":0}, {"x":7, "y":0}, {"x":8, "y":0}, {"x":9, "y":0}, {"x":10, "y":0}, {"x":11, "y":0},
+      			 {"x":0, "y":1}, {"x":1, "y":1}, {"x":2, "y":1}, {"x":3, "y":1}, {"x":4, "y":1}, {"x":5, "y":1}, {"x":7, "y":1}, {"x":8, "y":1}, {"x":9, "y":1}, {"x":10, "y":1}, {"x":11, "y":1},
+      			 {"x":0, "y":2}, {"x":1, "y":2}, {"x":2, "y":2}, {"x":3, "y":2}, {"x":4, "y":2}, {"x":5, "y":2}, {"x":7, "y":2}, {"x":8, "y":2}, {"x":9, "y":2}, {"x":10, "y":2}, {"x":11, "y":2},
+      			 {"x":0, "y":3}, {"x":1, "y":3}, {"x":2, "y":3}, {"x":3, "y":3}, {"x":4, "y":3}, {"x":5, "y":3}, {"x":7, "y":3}, {"x":8, "y":3}, {"x":9, "y":3}, {"x":10, "y":3}, {"x":11, "y":3},
+      			 {"x":0, "y":4}, {"x":1, "y":4}, {"x":2, "y":4}, {"x":3, "y":4}, {"x":4, "y":4},                                 {"x":8, "y":4}, {"x":9, "y":4}, {"x":10, "y":4}, {"x":11, "y":4}]
+    }
+  }
+}
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/info.json:16]:
```suggestion
{
    "keyboard_name": "freoduo",
    "url": "",
    "maintainer": "https://github.com/FilipParyz",
    "width": 12,
    "height": 5,
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"x": 0, "y": 0},
                {"x": 1, "y": 0},
                {"x": 2, "y": 0},
                {"x": 3, "y": 0},
                {"x": 4, "y": 0},
                {"x": 5, "y": 0},

                {"x": 7, "y": 0},
                {"x": 8, "y": 0},
                {"x": 9, "y": 0},
                {"x": 10, "y": 0},
                {"x": 11, "y": 0},
                {"x": 12, "y": 0},

                {"x": 0, "y": 1},
                {"x": 1, "y": 1},
                {"x": 2, "y": 1},
                {"x": 3, "y": 1},
                {"x": 4, "y": 1},
                {"x": 5, "y": 1},

                {"x": 7, "y": 1},
                {"x": 8, "y": 1},
                {"x": 9, "y": 1},
                {"x": 10, "y": 1},
                {"x": 11, "y": 1},
                {"x": 12, "y": 1},

                {"x": 0, "y": 2},
                {"x": 1, "y": 2},
                {"x": 2, "y": 2},
                {"x": 3, "y": 2},
                {"x": 4, "y": 2},
                {"x": 5, "y": 2},

                {"x": 7, "y": 2},
                {"x": 8, "y": 2},
                {"x": 9, "y": 2},
                {"x": 10, "y": 2},
                {"x": 11, "y": 2},
                {"x": 12, "y": 2},

                {"x": 0, "y": 3},
                {"x": 1, "y": 3},
                {"x": 2, "y": 3},
                {"x": 3, "y": 3},
                {"x": 4, "y": 3},
                {"x": 5, "y": 3},

                {"x": 7, "y": 3},
                {"x": 8, "y": 3},
                {"x": 9, "y": 3},
                {"x": 10, "y": 3},
                {"x": 11, "y": 3},
                {"x": 12, "y": 3},

                {"x": 0, "y": 4},
                {"x": 1, "y": 4},
                {"x": 2, "y": 4},
                {"x": 3, "y": 4},
                {"x": 4, "y": 4, "w": 2},

                {"x": 7, "y": 4, "w": 2},
                {"x": 9, "y": 4},
                {"x": 10, "y": 4},
                {"x": 11, "y": 4},
                {"x": 12, "y": 4}
            ]
        }
    }
}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by FilipParyz** [comment]
@fauxpark Thank you very much for the review, I'll fix the issues and get back here with next commit.

**Comment by FilipParyz** [comment]
@fauxpark All changes are up on the branch, qmk info works now, as it failed before.

File: keyboards/handwired/freoduo/rules.mk [hunk 10]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+# Bootloader selection
+BOOTLOADER = caterina
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/rules.mk:4]:
```suggestion

# Bootloader selection
BOOTLOADER = caterina

```

File: keyboards/handwired/freoduo/readme.md [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+A split 60%, 5x12 ortholinear keyboard handwired by Filip Paryż.
+
+Keyboard Maintainer: [Filip Paryz](https://github.com/FilipParyz)  
+Hardware Supported: Pro Micro, ATmega32u4
+
+Make example for this keyboard (after setting up your build environment):
+
+    make handwired/freoduo:default
+
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (fauxpark) [inline keyboards/handwired/freoduo/readme.md:13]:
```suggestion
# Freoduo

A split 60%, 5x12 ortholinear keyboard handwired by Filip Paryż.

* Keyboard Maintainer: [Filip Paryz](https://github.com/FilipParyz)
* Hardware Supported: Pro Micro, ATmega32u4

Make example for this keyboard (after setting up your build environment):

    make handwired/freoduo:default

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Since this is just a keyboard addition, it should go into master rather than develop, so it will be immediately available in the Configurator once merged.

**Comment by FilipParyz** [comment]
@fauxpark So should I make a new PR?

**Comment by fauxpark** [comment]
No, you can retarget this PR by hitting the Edit button at the top. You may need to rebase afterwards if a bunch of unrelated commits show up.

File: keyboards/handwired/freoduo/rules.mk (hunk 10 above)
Comment (FilipParyz) [reply keyboards/handwired/freoduo/rules.mk:4]: I don't understand this change. What changed?

**Comment by FilipParyz** [comment]
Had to do some switcharoo but it should work now.
All your suggestions (but one I commented on) should be in place.

File: keyboards/handwired/freoduo/rules.mk (hunk 10 above)
Comment (fauxpark) [reply keyboards/handwired/freoduo/rules.mk:4]: Added spacing around the block.

File: keyboards/handwired/freoduo/config.h [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x0616 //F is 6th in alphabet, P is 16th :)
```
Comment (drashna) [inline keyboards/handwired/freoduo/config.h:21]: Unfortunately, this VID is already in use. 
https://www.the-sz.com/products/usbid/index.php?v=0x0616

Please change this. ❯ 

For instance: 
```sh
echo -n "ParyzFilip" | shasum | cut -c1-4 | tr '[:lower:]' '[:upper:]'
```
produces: 
```suggestion
#define VENDOR_ID       0xB141
```
Which appears to be unused.

File: keyboards/handwired/freoduo/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,193 @@
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 3
```
Comment (drashna) [inline keyboards/handwired/freoduo/keymaps/default/keymap.c:10]: We recommend using an enum for tis instead. 

```suggestion
enum layers {
    _QWERTY,
    _LOWER,
    _RAISE,
    _ADJUST,
};
```

File: keyboards/handwired/freoduo/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+  _______, RGB_VAD, RGB_VAI, _______, _______,                   _______, _______, _______, _______, _______
+)
+
+
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/handwired/freoduo/keymaps/default/keymap.c:159]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

```suggestion
```

File: keyboards/handwired/freoduo/rules.mk (hunk 10 above)
Comment (drashna) [reply keyboards/handwired/freoduo/rules.mk:4]: The spacing. It adds a line between the MCU and bootloader section, adding a bit more readability.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by FilipParyz** [comment]
Thank you @drashna , I've made requested changes.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by FilipParyz** [comment]
Thank you @drashna  and @fauxpark  for your work for QMK :)

**Comment by drashna** [comment]
Thank you for your contribution!

**Pull request merged**